### PR TITLE
Generate static documentation pages & update the CI

### PR DESCRIPTION
**Pull request by GeorgeSittas**
Reference: https://kevsoft.net/2020/06/10/running-github-action-steps-and-jobs-only-on-push-to-master.html

### CODE REVIEW

**Comment by GeorgeSittas** [comment]
For some reason the python11 build skipped the integration tests.

Screenshot 2023-02-07 at 12 27 14 PM

**Comment by tobymao** [comment]
pyspark didn't work on py3.11 when it was first implemented

**Comment by GeorgeSittas** [comment]
Seems like the way I approached this won't work for persisting the docs in the repo. I'm searching for alternatives, feel free to make suggestions.

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -22,5 +22,17 @@ jobs:
         python -m pip install --upgrade pip
         make install-dev
     - name: Run checks (linter, code style, tests)
+      run: make check
+    - name: Update documentation
       run: |
-        make check
+        make docs
+        git add .
```
Comment (tobymao) [inline .github/workflows/python-package.yml:29]: git add docs

**Pull request merged**